src/Symfony/Bundle/FrameworkBundle/DependencyInjection/Compiler/TranslatorPass.php

```diff
@@ -40,6 +41,31 @@ public function process(ContainerBuilder $container)
             }
         }
 
-        $container->findDefinition('translator.default')->replaceArgument(2, $loaders);
+        $translatorDefinition = $container->findDefinition('translator.default');
+        $translatorDefinition->replaceArgument(2, $loaders);
+        if ($container->hasParameter('translator.resource_directories')) {
+            $resourceDirs = $container->getParameter('translator.resource_directories');
+            $files = array();
+            if ($resourceDirs) {
+                $finder = Finder::create()
+                    ->files()
+                    ->filter(function (\SplFileInfo $file) {
+                        return 2 === substr_count($file->getBasename(), '.') && preg_match('/\.\w+$/', $file->getBasename());
+                    })
+                    ->in($resourceDirs)
+                ;
+
+                foreach ($finder as $file) {
+                    list($domain, $locale, $format) = explode('.', $file->getBasename(), 3);
+                    if (!isset($files[$locale])) {
+                        $files[$locale] = array();
+                    }
+
+                    $files[$locale][] = (string) $file;
+                }
+            }
+
+            $translatorDefinition->replaceArgument(4, $files);
+        }
```

Comment:
What about moving this logic back into the extension instead? There is no need for it to be here and that would remove the need to create a parameter.